### PR TITLE
METRON-1136  Refactor for ParserExtensionConfig and Rest get all call

### PR DESCRIPTION
**Pull request by ottobackwards**
This a PR for some cleanup on the 1136 Feature Branch

- return a list from GET ( getAllParsers ) and not a map.
- fix names in Parser Extension Configs to be consistant with extension/extensions
- add a field just called identifier, to be used instead of assembly name
- differentiate identifier/assemblyname

## 
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron 
- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### CODE REVIEW

File: metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/ParserExtensionController.java [hunk 1]
```diff
@@ -90,8 +91,8 @@
   @ApiOperation(value = "Retrieves all ParserExtensionConfigs from Zookeeper")
   @ApiResponse(message = "Returns all ParserExtensionConfigs", code = 200)
   @RequestMapping(method = RequestMethod.GET)
-  ResponseEntity<Map<String, ParserExtensionConfig>> findAll() throws RestException {
-    return new ResponseEntity<Map<String, ParserExtensionConfig>>(extensionService.getAllParserExtensions(), HttpStatus.OK);
+  ResponseEntity<List<ParserExtensionConfig>> findAll() throws RestException {
```
Comment (nickwallen) [inline metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/ParserExtensionController.java]: What is the logic for changing this from a map to a list?

File: metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/impl/ExtensionServiceImpl.java [hunk 2]
```diff
@@ -528,20 +527,26 @@ private void rollBackPatternsFromHdfs(InstallContext context) throws Exception{
...
     config.setExtensionAssemblyName(context.extensionPackageName.get());
     config.setDefaultParserConfigs(context.defaultParserConfigs.get());
-    config.setDefaultEnrichementConfigs(context.defaultEnrichmentConfigs.get());
+    config.setDefaultEnrichmentConfigs(context.defaultEnrichmentConfigs.get());
     config.setDefaultIndexingConfigs(context.defaultIndexingConfigs.get());
     if(context.defaultElasticSearchTemplates.isPresent()) {
       config.setDefaultElasticSearchTemplates(context.defaultElasticSearchTemplates.get());
     }
-    ConfigurationsUtils.writeParserExtensionConfigToZookeeper(context.extensionPackageName.get(),config.toJSON().getBytes(), client);
+    ConfigurationsUtils.writeParserExtensionConfigToZookeeper(context.extensionIdentifier.get(),config.toJSON().getBytes(), client);
```
Comment (nickwallen) [inline metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/impl/ExtensionServiceImpl.java]: What is the logic for changing package name to identifier?

File: metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/ParserExtensionController.java (hunk 1 above)
Comment (ottobackwards) [reply metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/ParserExtensionController.java]: I have another branch where I am doing work on extending the management ui to manage extensions.  I had a lot of trouble consuming the maps.  The map is the configurations with a field pulled out as the key, so making it a list made it work, similar to the getAll for parser configurations, and really didn't lose anything.

TL;DR
I am not good enough at angular to consume the maps correctly so I refactored.

File: metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/impl/ExtensionServiceImpl.java (hunk 2 above)
Comment (ottobackwards) [reply metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/impl/ExtensionServiceImpl.java]: While working with the config, the fact that I had picked a field to use as the identifier, but was also something else seemed confusing.  Making the identifier explicit, and also making that field what it should be ( the assemblyName/packageName ) without truncating it seems to me to make it more understandable.

File: metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/ParserExtensionController.java (hunk 1 above)
Comment (nickwallen) [reply metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/ParserExtensionController.java]: Makes sense.  Its cleaner this way.  (Even if the real reason is that you aren't good enough with Angular.  Ha)

**Comment by nickwallen** [comment]
+1 Thanks, Otto

**Comment by anandsubbu** [comment]
Hi @ottobackwards , I guess you wanted to name the PR title as METRON-1136 instead of METRON-1146?

**Comment by ottobackwards** [comment]
I just got done doing that!  Luckily the scripts that I use have you verify everything, and I caught the different message :)
